design/one-pager-stacks-versioning.md

```diff
@@ -0,0 +1,248 @@
+https://s3.amazon.com/mybucket/mychannel/all # All version info
+
+The stack installer would initially only use https://s3.amazon.com/mybucket/mychannel/latest, but in the future if
+ the user set an upgrade policy it would likely read from all if necessary depending on rules.
+
+### Installing from stable
+
+The stack installer will be version aware, namely it will understand either how to upgrade to the tip of a release
+ channel, or how to follow a pre-defined upgrade policy by reading the entire list of available versions.
+
+Consider the following scenario where the installer wishes to auto-upgrade to the lastest version. If the current
+ version is 1.2.3 this would be reflected in the stack install status and the package would be set to
+ `crossplane/aws-stack`.  The channel would be set to the stable release channel url hosted by s3 or other means with
+ the following valid payload from `/latest` from said channel.
+
+```json
+{
+ "name": "stable",
+ "type": "channel",
+ "package": "crossplane/aws-stack",
+ "latest":
+  {
+    "version": "1.2.3",
+    "id": "461324714c7d",
+    "createTime": "2019-09-12T17:39:04Z"
+  }
+}
+```
+
+To enact an upgrade in this scenario, you would update the payload to be a higher semantic version, i.e
+ `{"latest": {"version": 1.2.4", ...} ... }`. The installer would then install a new stack image from
+ `crossplane/aws-stack:shaofversion` by looking up the actual underlying artifact for that version in the channel.
+ The stack installer status should reflect that is installing 1.2.4, with the particular sha.
+
+If you update your release channel to something of a lesser value the stack installer would return an error that release
+ channel is of a lesser version and not upgrade. Further, if we pushed a version that contained pre-release info, this
+ would not match as a stable version and would also fail to update. Ideally we would disallow pushing invalid versions
+ to channels as part of our internal tooling.
+
+### Arbitrary channel support
+
+Stable is a reserved channel name to indicate clean semantic versions with no pre-release metadata. Any arbitrary
+ lowercase word can be used to denote a channel name and must appear as the first item in the pre-release metadata.
+
+Example of the beta channel:
+
+```json
+ {
+  "name": "beta",
+  "type": "channel",
+  "package": "crossplane/aws-stack",
+  "latest":
+   {
+     "version": "1.2.3-beta+123",
+     "id": "461324714c7d",
+     "createTime": "2019-09-12T17:39:04Z"
+   }
+ }
+ ```
+
+### Support for complex upgrades with a list of versions + update policy
+
+Future work could further build on the latest concept in our release channels and instead define update policy.
+ Instead of forcing the stack to only update to the next version, we will support maintenance and upgrade policies.
+
+An example: Imaging the following upgrade policy - Apply patch versions immediately, minor versions in a pre-defined
```

Comment:
What are the considerations for pull-driven release channel upgrades versus push-driven release channel upgrades? Considerations such as scaling request volume, and servicing control clusters which are not accessible to be pushed to. Are we assuming pull-driven here?